### ISSUE TITLE
Training speed

### ISSUE THREAD

**Issue by yongmayer** [Aradhye2002/EcoDepth#11]
Hello,

Thank you so much sharing this amazing work! I wonder how fast to train the model (for nyu dataset) per epoch? Say, I am using a single A100 (80GB) GPU with batch-size 32. In my case, it seems stuck in the following step for ever...

`        with torch.no_grad():
            # convert the input image to latent space and scale.
            latents = self.encoder_vq.encode(x).mode().detach() * self.config.model.params.scale_factor
`

I would appreciate any help/hint. Again, thank you so much!

P.S., I can run the evaluation job successfully with results as follows. It matches with the paper well except for sq_rel.
========================================================================================
        d1         d2             d3        abs_rel     sq_rel       rmse     rmse_log    log10      silog 
    0.9776     0.9973     0.9995     0.0599     0.0194     0.2187     0.0773     0.0259     5.7549 
========================================================================================

**Comment by Aradhye2002** [comment]
Hi @yongmayer, thanks for appreciating our work. So we used a per device batch size of 4 resulting in a total batch size of 32 with 8 GPUs. The speed issue is probably because you are using a per device batch size of 32 instead of 4. Could you try once with a batch size of 4 (with a single GPU ie. you current setup) and let me know if it works?

**Comment by yongmayer** [comment]
Thank you so much! That works!

I have another question if you don't mind asking. How should I understand the diffusion process in EcoDepth? 
From line 96 in EcoDepth/depth/models/model.py (EcoDepthEncoder.forward), I see it uses the Unet from stable diffusion, but cannot see how the forward diffusion process. Am I understanding it wrong? I am new to the diffusion-based depth estimation, and I would appreciate your kind explanation a lot! 

Again, thank you!

**Comment by Aradhye2002** [comment]
Hi @yongmayer. While we do use the stable diffusion backbone, we do not use it as a diffusion model per se. Instead we use the UNet (or backbone) as a feature extractor and utilize the hierarchical feature maps obtained to predict the final depth map. Please refer to our model architecture given in the paper for more details.
